x-pack/plugins/security_solution/public/detections/components/rules/mitre/helpers.test.tsx

```diff
@@ -4,10 +4,146 @@
+      });
+
+      it('returns true if tacticName exists and is not `none`', () => {
+        expect(isMitreAttackInvalid('Test', [])).toBe(true);
+      });
+    });
+
+    describe('when technique param exists', () => {
+      describe('and contains valid techniques', () => {
+        const validTechniques = [
+          {
+            reference: 'https://test.com',
+            name: 'Audio Capture',
+            id: 'T1123',
+          },
+        ];
+        it('returns false', () => {
+          expect(isMitreAttackInvalid('Test', validTechniques)).toBe(false);
+        });
+      });
+
+      describe('and contains empty techniques', () => {
+        const emptyTechniques = [
+          {
+            reference: 'https://test.com',
+            name: 'none',
+            id: '',
+          },
+        ];
+        it('returns true', () => {
+          expect(isMitreAttackInvalid('Test', emptyTechniques)).toBe(true);
+        });
+      });
+    });
+  });
+
+  describe('isMitreTechniqueInvalid', () => {
+    describe('when technique param is undefined', () => {
+      it('returns false if tacticName is `none`', () => {
+        expect(isMitreTechniqueInvalid('none', undefined)).toBe(false);
+      });
+
+      it('returns false if tacticName exists and is not `none`', () => {
+        expect(isMitreTechniqueInvalid('Test', undefined)).toBe(false);
+      });
+    });
+
+    describe('when technique param exists', () => {
+      describe('and has possible subtechnique options', () => {
+        describe('and contains at least one of those options', () => {
+          const validTechnique = {
+            reference: 'https://test.com',
+            name: 'Archive Collected Data',
+            id: 'T1560',
+            subtechnique: [
+              { reference: 'https://test.com', name: 'Archive via Library', id: 'T1560.002' },
+            ],
+          };
+          it('returns false', () => {
+            expect(isMitreTechniqueInvalid('Test', validTechnique)).toBe(false);
+          });
+        });
+
+        describe("and doesn't contain any subtechniques", () => {
+          const validTechnique = {
+            reference: 'https://test.com',
+            name: 'Archive Collected Data',
+            id: 'T1560',
+            subtechnique: [],
+          };
+          it('returns true', () => {
+            expect(isMitreTechniqueInvalid('Test', validTechnique)).toBe(true);
+          });
+        });
+
+        describe('and contains only empty subtechniques', () => {
+          const validTechnique = {
+            reference: 'https://test.com',
+            name: 'Archive Collected Data',
+            id: 'T1560',
+            subtechnique: [{ reference: '', name: 'none', id: '' }],
+          };
+          it('returns true', () => {
+            expect(isMitreTechniqueInvalid('Test', validTechnique)).toBe(true);
+          });
+        });
+      });
+
+      describe("and doesn't have possible subtechnique options", () => {
```

Comment:
These tests are a bit brittle because they assume that the mitre data doesn't change. If at some point the `Audio Capture` technique changes, for example, some of these tests would fail even though the helper function code hasn't changed and is still correct. You could make these more stable by mocking some of the underlying functions that the helpers use.